src/snakemake/rules.py

```
@@ -474,6 +475,9 @@ def _set_inoutput_item(self, item, output=False, name=None, mark_ancient=False):
 
             item = default_flags.apply(item)
 
+            for flag in self.workflow.storage_settings.omit_flags:
```

Flake8 (7.2.0)

[error] import 'flag' from line 23 shadowed by loop variable

(F402)